x/blob/ante/ante.go

```diff
@@ -0,0 +1,65 @@
+package ante
+
+import (
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+	"github.com/celestiaorg/celestia-app/pkg/shares"
+	"github.com/celestiaorg/celestia-app/x/blob/types"
+
+	"cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// MinGasPFBDecorator helps to prevent a PFB from being included in a block
+// but running out of gas in DeliverTx (effectively getting DA for free)
+// This decorator should be run after any decorator that consumes gas.
+type MinGasPFBDecorator struct {
+	k BlobKeeper
+}
+
+func NewMinGasPFBDecorator(k BlobKeeper) MinGasPFBDecorator {
+	return MinGasPFBDecorator{k}
+}
+
+// AnteHandle implemnts the AnteHandler interface. It checks to see
+// if the transaction contains a MsgPayForBlobs and if so, checks that
+// the transaction has allocated enough gas.
+//
+// TODO: We need to run this antehandler in process proposal
+// however this will be a breaking change.
+func (d MinGasPFBDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	if !ctx.IsCheckTx() {
+		return next(ctx, tx, simulate)
+	}
```

Comment:
no #1985 ?

Reply:
https://github.com/celestiaorg/celestia-app/pull/1985#pullrequestreview-1495014460

Reply:
Will add it now